src/components/NavBar/NavBar.tsx

```diff
@@ -1,9 +1,36 @@
-import React, { useState } from 'react';
+import React, { useState ,useContext } from 'react';
 import './NavBar.scss';
 import Logo from './../images/logo.png';
 import {NavLink} from 'react-router-dom';
 import Flex from './../Flex/Flex';
+import { UserContext } from '../../pages/HomePage/HomePage';
 const NavBar = () => {
+  const { state, dispatch } = useContext(UserContext);
+  const RenderMenu = () =>{
+    if(state){
+      return(
+        <ul>
+          <li ><NavLink exact to="/" >Home</NavLink></li>
+          <li ><NavLink to="/postjobs">Post a Job</NavLink></li>
+          <li ><NavLink to="/contactus">Contact Us</NavLink></li>
+          <li > <NavLink to="/payment">Payment</NavLink></li>
+          <li ><NavLink to="/logout">Logout</NavLink></li>
```

Comment:
only logout, login parts are changing right? you can add conditional rendering only for that. you can try that with ternary operator.